### PR TITLE
feat: Angular 14

### LINKED ISSUE #90: Update to Angular 14
We should probably start matching our major versions to the Angular major versions as well...

### PR DESCRIPTION
**Pull request by bobbyg603**
### Description

Updates sample to Angular 14. Updates ng-packagr. Ensures Angular 14 compatibility. Fixes linting.

Resolves #90 

### Checklist

- [x] Tested manually
- [x] Unit tests pass with no errors or warnings
- [x] Documentation updated (if applicable)
- [x] Reviewed by at least 1 other contributor

### CODE REVIEW

File: projects/my-angular-crasher/src/app/app.component.ts [hunk 1]
```diff
@@ -48,7 +48,8 @@ export class AppComponent implements OnInit {
     this.link$ = bugsplat.getObservable()
       .pipe(
         map((bugSplatEvent) => {
-          const crashId = bugSplatEvent.responseData.crash_id;
+          // TODO BG test
+          const crashId = bugSplatEvent.responseData.crashId;
```
Comment (bobbyg603) [inline projects/my-angular-crasher/src/app/app.component.ts:52]: TODO BG

Comment (bobbyg603) [reply projects/my-angular-crasher/src/app/app.component.ts:51]: Resolved in [12cca6a](https://github.com/BugSplat-Git/bugsplat-ng/pull/93/commits/12cca6acda73de0ff3c4e8f460150247305fb8a4)

File: package.json [hunk 2]
```diff
@@ -58,30 +57,32 @@
...
-    "typescript": "~4.5.5",
-    "semantic-release": "^19.0.3"
+    "typescript": "~4.7.4"
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/BugSplat-Git/bugsplat-ng.git"
   }
-}
+}
```
Comment (zmrl010) [inline package.json:88]: seeing a few of these no new lines. not sure if that is important to you

**Review by zmrl010 — APPROVED** [review]
Everything looks good for the most part!

I noticed there were a couple files with no ending blank line, but that is mostly a nitpick

File: package.json (hunk 2 above)
Comment (bobbyg603) [reply package.json:88]: tried fixing this but since i couldn't get it within a few minutes i decided to bail!

**Pull request merged**